mrT.go

```diff
@@ -202,29 +224,10 @@ func (m *MrT) isInCurrent(txnID string) (ok bool) {
 	return ru.Time().UnixNano() >= pu.Time().UnixNano()
 }
 
-func (m *MrT) flush() (err error) {
-	// Ensure we are at the end before flushing
-	if err = m.s.SeekToEnd(); err != nil {
-		return
-	}
-
-	_, err = io.Copy(m.f, m.buf)
-	m.buf.Reset()
-	return
-}
-
-func (m *MrT) rollback() {
-	m.buf.Reset()
-}
-
 func (m *MrT) readArchiveLines(fn func(*bytes.Buffer) error) (err error) {
-	var af file.Interface
-	if af, err = os.Open(path.Join(m.dir, "archive", m.name+".tdb")); err != nil {
-		return
-	}
-	defer af.Close()
-
-	as := seeker.New(af)
+	ar := m.af.Reader()
+	defer ar.Close()
+	as := seeker.New(ar)
 	defer as.SetFile(nil)
```

Comment:
what's the point of this call?

Reply:
It's just to clear the file from the seeker for easier garbage collection

Reply:
This is 100% not needed and the overhead of a defer is probably much higher than the uber-tiny-micro-tiny optimization.